### ISSUE TITLE
parsing fonts.googleapis.com incorrect

### ISSUE THREAD

**Issue by amitmtrn** [peerigon/parse-domain#4]
parsing `fonts.googleapis.com` isn't work correctly. the subdomain should be `fonts` the domain should be `googleapis` and the tld should be `com`.
this is what I got while parsing the domain
`{ tld: 'googleapis.com', domain: 'fonts', subdomain: '' }`

**Comment by jhnns** [comment]
Mhmm ... that's tricky, because `googleapis.com` is listed at [publicsuffix](https://publicsuffix.org/list/effective_tld_names.dat), since the "user controlled" part of the URL is placed before `googleapis.com`. Same with `blogspot.com`.

That's why `parseDomain()` thinks it's a TLD although it's technically not a TLD. I probably used the term TLD in a wrong way, since a TLD is *always* the last portion of the URL (like `uk` in `.co.uk`). However, most users care more about the "user controlled" part which can be very subjective depending on your use-case.

Browser vendors, however, use [publicsuffix as source to determine whether the entered string is a URL or a search keyword and even for restricting cookie access](https://publicsuffix.org/learn/).

Honestly, I don't know how to proceed. Can we define an expected behavior? Maybe publicsuffix is the wrong source, maybe we should leverage DNS?

**Comment by amitmtrn** [comment]
That's interesting, I didn't know the concept of public suffix. I used this module to separate the domain name and use whois to check who own this domain. I think that maybe it would be best to add the domain LTD and the domain publicsuffix.

for example:
`something.blogspot.com` whould be parse to
```
{
domain: blogspot,
subdomain: something,
LTD: com,
publicsuffix: blogspot.com 
}
```

**Comment by hongkongkiwi** [comment]
Seems I am having the same issue as above. It would be great to seperate out publicsuffix, because honestly that's just a guess and we will never catch them all. Where as tld is very clear, there can only be a finite number of tld's and in my case I'm most interested in the tld, domain, subdomain breakdown.

**Comment by jhnns** [comment]
Well, as far as I can tell, is the TLD just the last portion of the URL. I don't know if there is a rule that separates the `.co.uk` case from the `blogspot.com`. That's why publicsuffix.com calls itself a *"A list of effective TLDs"*.

However, since browser vendors use this list even for restricting cookie access, it's not just an arbitrary list. But I have to admit that it's somewhat surprising...

I'm thinking about using DNS resolution to distinct between `co.uk` and `blogspot.com`. Former returns `Can't find co.uk: No answer` while the latter returns a IP.

**Comment by jhnns** [comment]
parse-domain excludes private domains by default now. Shipped with `1.0.0`.
